src/data_structures/arrays/ReverseString.java

```diff
@@ -15,8 +15,17 @@ public static String ReverseStringWithReelJava(String string) {
     return new StringBuilder(string).reverse().toString();
   }
 
+  public static String recursiveReversal (String str) {
+    if (str.length() == 0) {  //breakpoint for recursion
+       return "";
+    } else {
```

Comment:
`else` is redundant here because of the `return` keyword.

Reply:
No. It is not. This is a recursive function.  Return will happen only when str.length() == 0.

Reply:
Yep, the first `return` will happen only when `str.length()` is `0` but when this condition is not met it would go to the next line anyway, regardless of the fact that this is a recursive function. 
I suggest running with the debugger on this to see why it's behaving the same.

This of course isn't about the logic, just about the syntax...

Reply:
please comment your code change. will take a look